helpdesk/views/staff.py

```diff
@@ -508,20 +497,14 @@ def get_ticket_from_request_with_authorisation(
                 is_helpdesk_staff(request.user) or
                 helpdesk_settings.HELPDESK_ALLOW_NON_STAFF_TICKET_UPDATE))):
 
-        key = request.POST.get('key')
-        email = request.POST.get('mail')
-
-        if key and email:
-            ticket = Ticket.objects.get(
+        try:
+            return Ticket.objects.get(
                 id=ticket_id,
-                submitter_email__iexact=email,
-                secret_key__iexact=key
-            )
-
-        if not ticket:
-            return HttpResponseRedirect(
-                '%s?next=%s' % (reverse('helpdesk:login'), request.path)
+                submitter_email__iexact=request.POST.get('mail'),
+                secret_key__iexact=request.POST.get('key')
             )
+        except (Ticket.DoesNotExist, ValueError):
+            return redirect_to_login(request.path, 'helpdesk:login')
 
```

Comment:
This change may not be ideal.
It should probably check if "key" and "email" are in the request and retiurn a different error message than just redirecting to the login page. 
I am not familiar with the UI usage of this form so it may be acceptable.

Reply:
Indeed, I simply refactored this part of the code but an error handling would be better. Maybe I can only catch `Ticket.DoesNotExist` here and let the ValueError be handled in the calling view (which is `update_ticket`) ?

Reply:
does `update_ticket` do any error catching for ValueError? I don't have time at the moment to investigate.

Reply:
No it is not currently handling this type of Exception. But I'm still wondering what would be the best. It seems to be easier to do error handling directly in that view since we have access to the request in the parameter

Reply:
I think you're probably right, it just means we might need to do some extra work on `update_ticket` too as part of this PR. thanks!

Reply:
Maybe something frontend with an alert?